server.py

```diff
@@ -246,8 +246,30 @@ def get(self, path, feature_id=None):
     self.render(data=template_data, template_path=os.path.join(path + '.html'))
 
 
+class FeaturesAPIHandler(common.JSONHandler):
+
+  def __get_feature_list(self, version=None):
+    feature_list = models.Feature.get_chronological(version=version) # Memcached
+    annotate_first_of_milestones(feature_list, version=version)
+    return feature_list
+
+  def get(self, version=None):
+    if version is None:
+      version = 1
+    else:
+      version = int(version[2:])
+
+    KEY = '%s|v%s|all' % (models.Feature.DEFAULT_MEMCACHE_KEY, version)
+    feature_list = memcache.get(KEY)
+    if feature_list is None:
+      feature_list = self.__get_feature_list(version)
+      memcache.set(KEY, feature_list)
+    return common.JSONHandler.get(self, feature_list, formatted=True)
+
+
 # Main URL routes.
 routes = [
+  ('/features(_v[0-9]*)?.json', FeaturesAPIHandler),
```

Comment:
I guess this feeds into the previous comment I had, an explains why you want to discard the first two characters of the `version` value.

How about changing the regular expression to exclude the `_v` from the capture group and ensure that there's always a digit whenever there's a `_v`?

```py
(r'/features(?:_v(\d+))?.json', FeaturesAPIHandler)
```